### PR TITLE
spectre-and-meltdown-on-apu2.md: initial version of the post

### PR DESCRIPTION
**Pull request by miczyg1**
Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

### CODE REVIEW

**Comment by miczyg1** [comment]
@pietrushnic preceding post with spectre and meltdown intro is still missing.

File: _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+```
+
+After microcode patching:
+
+```
+* PRED_CMD MSR is available:  YES
+* CPU indicates IBPB capability:  YES  (IBPB_SUPPORT feature bit)
+...
+* IBPB enabled and active:  YES
+```
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: @pietrushnic sure. Where should we store graphics for the posts?

File: _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+
+Now load msr module, give execution permission to the script and run it:
+
+```
+chmod +x spectre-meltdown-checker.sh
+sudo modprobe msr
+sudo ./spectre-meltdown-checker.sh
+```
+
+[output is quite long, should i paste it all?]
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: @pietrushnic almost all information is critical. I just can avoid pasting Intel related vulnerabilities, like SGX etc.

File: _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+    - Firmware
+    - Security
+    - BIOS
+
+
+---
+
+# Spectre and meltdown
+
+In the [previous post](Meltdown Spectre intro TBD) I have introduced the
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: @pietrushnic maybe we should add him for review? But he seems not to be in organization

File: _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (miczyg1) [reply _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: @pietrushnic I have replaced the output with two images. Also some information is cut.

File: _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (miczyg1) [reply _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: NVM. I have pasted the script output as a graphics above

[the remark this answers is not in the document]
Comment (miczyg1) [reply _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: @Moviss I have pasted an embedded player for asciinema. Does our blog support launching such scripts?

**Comment by artur-rs** [comment]
@pietrushnic @miczyg1  this PR can be closed?

File: _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pietrushnic) [reply _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: @artur-rs this is good question in light of Hugo

**Comment by pietrushnic** [comment]
@artur-rs this blog post was not yet published it is IMO ready to be published. @miczyg1 ?

**Comment by miczyg1** [comment]
@artur-rs the post needs Yoast/SEO optimizations for better readability and also some fixes to be consumed by Hugo. I plan to finish it in this release

File: _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (artur-rs) [reply _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: @miczyg1 yes, it is supported. Syntax should look like this:
```
[![asciicast](https://asciinema.org/a/xJC0QaKuHrMAPhhj5KMZUhMEO.svg)](https://asciinema.org/a/xJC0QaKuHrMAPhhj5KMZUhMEO?speed=1)
```

File: _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+    - meltdown
+    - apu2
+    - PC Engines
+    - coreboot
+categories:
+    - Firmware
+    - Security
+    - BIOS
+
+
```
Comment (artur-rs) [inline _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: outdated header - then newest one is located in `blog/content/post/YYYY-MM-DD*` file

File: _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+chmod +x spectre-meltdown-checker.sh
+sudo modprobe msr
+sudo ./spectre-meltdown-checker.sh
+```
+
+The result is as follows
+
+1. Hardware support for mitigation techniques:
+
+![Hardware support (CPU microcode) for mitigation techniques](https://3mdeb.com/wp-content/uploads/2017/07/spectre_hw.png)
```
Comment (artur-rs) [inline _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: WP links are outdated

**Comment by artur-rs** [comment]
@miczyg1 parent branch should be set to `develop`. All commits must be `verified`, otherwise PR cannot be merged.

**Comment by miczyg1** [comment]
@artur-rs thank You for review, I will address the discussions.

File: _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (miczyg1) [reply _posts/2019-01-XX-spectre-and-meltdown-on-apu2.md]: I have added graphics and embedded asciinema cast. Marking as resolved.

**Comment by miczyg1** [comment]
@pietrushnic post converted to Hugo. If You could go through the post and make the review again please, it would be great. We have already scheduled it for 29th of May.

File: blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+
+---
+
+# Meltdown and Spectre
+
+In the [previous post](https://blog.3mdeb.com/2019/2019-03-14-meltdown-and-spectre-vulnerabilities/)
+I have introduced the Meltdown and Spectre vulnerabilities of modern x86
+processors and what threat do they pose to security and safety of the data. As
+a continuation of the last post, I will demonstrate the state of Meltdown and
+Spectre vulnerabilities on PC Engines apu2 platforms. We have recently added a
```
Comment (pietrushnic) [inline blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md]: @miczyg1 recently?
```suggestion
Spectre vulnerabilities on PC Engines apu2 platforms. Some time ago we have added a
```

File: blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+microcode update feature to PC Engines firmware so I will show how microcode
+update improves mitigation and present results using public tools/Proof of
```
Comment (pietrushnic) [inline blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md]:
```suggestion
update improves mitigation and present results using public tools/proof of
```

File: blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+
+After trying to run the PoC with 20/70/100 the string is not disclosed. But
+what if the kernel we use does not have the mitigation? The answer is firmware
+and microcode for full hardware mitigation.
+
+# Microcode update and spectre
+
+To present results of Spectre vulnerability without kernel mitigation, we will
+need the microcode update. For details how to build PC Engines apu2 coreboot
+firmware with microcode update, please refer to this [guide](https://github.com/pcengines/apu2-documentation/blob/master/docs/microcode_patching.md)
```
Comment (pietrushnic) [inline blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md]: @miczyg1 does this procedure still work fine? There are some bug reports related: https://github.com/pcengines/apu2-documentation/issues/145

File: blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+combined mitigation in firmware/microcode and kernel give the best protection.
+Although microcode update seems to not improve the situation in this particular
+case.
+
+I hope this post was useful for you. Please feel free to share your opinion and
+if you think there is value, then share with friends.
+
+If you think we can help in improving the security of your firmware or you
+looking for someone who can boot your product by leveraging advanced features
+of used hardware platform, feel free to [boot a call with us](https://calendly.com/3mdeb/consulting-remote-meeting)
```
Comment (pietrushnic) [inline blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md]:
```suggestion
of used hardware platform, feel free to [book a call with us](https://calendly.com/3mdeb/consulting-remote-meeting)
```

File: blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md (hunk 8 above)
Comment (miczyg1) [reply blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md]: @pietrushnic this bug is related to docker itself not to build process. I have an asciinema in the pst showing how to build the image. Worked perfectly fine for the version mentioned in blogpost.

File: blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md (hunk 6 above)
Comment (miczyg1) [reply blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md]: Done

File: blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md (hunk 7 above)
Comment (miczyg1) [reply blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md]: Done

File: blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md (hunk 9 above)
Comment (miczyg1) [reply blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md]: Done

File: blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md (hunk 8 above)
Comment (miczyg1) [reply blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md]: Also checked with new v4.9.0.5. No problems with building and flashing. For mainline releases we use unmodified coreboot-sdk container, so it is the Linxu configuration problem with permissions. It is highly dependable on groups, directory configuration, permissions, structure. May even be a file system (like if someone would like to compile coreboot on NTFS for some reason). Anyway, I have built is successfully without problems, waiting for the response to my comment in linked issue.

Comment (pietrushnic) [reply blog/content/post/2019-05-29-spectre-and-meltdown-on-apu2.md]: Ok

**Comment by artur-rs** [comment]
@miczyg1 reminder: post is scheduled on this week. Last fixes for Hugo archives requires new taxonomy metadata in the post header, please add (according to template):
```
archives: "2019"
```

**Comment by miczyg1** [comment]
@artur-rs done

**Pull request merged**